pilot/pkg/xds/debug.go

```diff
@@ -193,10 +194,35 @@ func (s *DiscoveryServer) AddDebugHandlers(mux *http.ServeMux, enableProfiling b
 	s.addDebugHandler(mux, "/debug/networkz", "List cross-network gateways", s.networkz)
 }
 
+func DebugFilter(handlerF func(http.ResponseWriter, *http.Request)) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		addr, _, err := net.SplitHostPort(r.RemoteAddr)
+		if err != nil {
+			_, _ = w.Write([]byte("invalid request"))
+		}
+		fromLocal := false
+		if addr == "localhost" {
```

Comment:
RemoteAddr is always an IP

Reply:
There is localhost check in pilot-agent status server.  
https://github.com/istio/istio/blob/master/pilot/cmd/pilot-agent/status/server.go#L348~L356

Reply:
+1, can you please switch to this?